core/src/main/java/hudson/model/ChoiceParameterDefinition.java

```diff
@@ -141,21 +141,27 @@ public StringParameterValue getDefaultParameterValue() {
         return new StringParameterValue(getName(), defaultValue, getDescription());
     }
 
-    private StringParameterValue checkValue(StringParameterValue value) {
-        if (!choices.contains(value.value))
-            throw new IllegalArgumentException("Illegal choice for parameter " + getName() + ": " + value.value);
-        return value;
+    @Override
+    public boolean isValid(ParameterValue value) {
+        return choices.contains(((StringParameterValue) value).getValue());
     }
 
     @Override
     public ParameterValue createValue(StaplerRequest req, JSONObject jo) {
         StringParameterValue value = req.bindJSON(StringParameterValue.class, jo);
         value.setDescription(getDescription());
-        return checkValue(value);
+        if (!isValid(value)) {
+            throw new IllegalArgumentException("Illegal choice for parameter " + getName() + ": " + value.getValue());
+        }
```

Comment:
You can extract this piece of code keeping the `checkValue` method but using the new `isValid` method. This way you avoid repeating code below in the `createValue(String)` method.